### PR TITLE
Add black formatter to pre-commit config

### PR DESCRIPTION
**Pull request by LukeHankey**
Currently set the line length to 100. I feel 88 (default) can sometimes wrap code in an odd way and 100 is just a nice round number. Open to changes.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -29,3 +29,11 @@ repos:
...
         additional_dependencies:
           - flake8-docstrings~=1.6.0
+
+  - repo: https://github.com/psf/black
+    rev: 22.6.0
+    hooks:
+      - id: black
+        language_version: "3.10"
+        args:
+          - --line-length=119
```
Comment (kappq) [inline .pre-commit-config.yaml:39]: I have tested it on EndeavourOS and it works great. However, if the `flake8` tests pass, formatting isn't needed. Though it is still useful when the tests fail.

**Review by kappq — APPROVED** [review]

File: .pre-commit-config.yaml (hunk 1 above)
Comment (LukeHankey) [reply .pre-commit-config.yaml:39]: Does flake8 check for everything that black does? [Here](https://flake8.pycqa.org/en/latest/user/error-codes.html) is the list of default codes that we get with installing flake8, as well as #2 to come.

Comment (kappq) [reply .pre-commit-config.yaml:39]: I am not sure, but `black` is useful anyway, so we can just keep it.

**Review by bensgilbert — APPROVED** [review]
looks good

**Pull request merged**